astropy/visualization/mpl_normalize.py

```diff
@@ -212,6 +212,231 @@ def inverse(self, values, invalid=None):
+        level will set at the ``(100 - percent) / 2`` percentile, while
+        the upper cut level will be set at the ``(100 + percent) / 2``
+        percentile. The default is 100.0. ``percent`` is ignored if
+        either ``min_percent`` or ``max_percent`` is input.
+
+    min_percent : float, optional
+        The percentile value used to determine the pixel value of
+        minimum cut level. The default is 0.0. ``min_percent`` overrides
+        ``percent``.
+
+    max_percent : float, optional
+        The percentile value used to determine the pixel value of
+        maximum cut level. The default is 100.0. ``max_percent``
+        overrides ``percent``.
+
+    vmin : float, optional
+        The pixel value of the minimum cut level. Data values less
+        than ``vmin`` will set to ``vmin`` before stretching the
+        image. The default is the image minimum. ``vmin`` overrides
+        ``min_percent``.
+
+    vmax : float, optional
+        The pixel value of the maximum cut level. Data values greater
+        than ``vmax`` will set to ``vmax`` before stretching the
+        image. The default is the image maximum. ``vmax`` overrides
+        ``max_percent``.
+
+    power : float, optional
+        The power index for ``stretch='power'``. The default is 1.0.
+
+    log_a : float, optional
+        The log index for ``stretch='log'``. The default is 1000.
+
+    asinh_a : float, optional
+        For ``stretch='asinh'``, the value where the asinh curve
+        transitions from linear to logarithmic behavior, expressed as a
+        fraction of the normalized image. Must be in the range between 0
+        and 1. The default is 0.1.
+
+    sinh_a : float, optional
+        The scaling parameter for ``stretch='sinh'``. The default is
+        0.3.
+
+    clip : bool, optional
+        If `True`, data values outside the [0:1] range are clipped to
+        the [0:1] range.
+
+    invalid : None or float, optional
+        Value to assign NaN values generated by the normalization. NaNs
+        in the input ``data`` array are not changed. For matplotlib
+        normalization, the ``invalid`` value should map to the
+        matplotlib colormap "under" value (i.e., any finite value < 0).
+        If `None`, then NaN values are not replaced. This keyword has no
+        effect if ``clip=True``.
+
+    See Also
+    --------
+    simple_norm
+
+    Examples
+    --------
+    .. plot::
+        :include-source:
+
+        import numpy as np
+        import matplotlib.pyplot as plt
+        from astropy.visualization import SimpleNorm
+
+        image = np.arange(65536).reshape((256, 256))
+        snorm = SimpleNorm('sqrt', percent=98)
+        norm = snorm(image)
+        fig, ax = plt.subplots()
+        axim = ax.imshow(image, norm=norm, origin='lower')
+        fig.colorbar(axim)
+    """
+
+    def __init__(
+        self,
+        stretch="linear",
+        percent=None,
```

Comment:
Let's make every argument beyond `stretch` keyword-only from the get go
```suggestion
        *,
        percent=None,
```

Reply:
I'd prefer to make it keyword-only after `percent`.  99.99% of the time I only use `percent` interactively and that would save me a lot of typing.  :smile: And I use this functionality a lot!

Addressing your other comments now, thanks!

Reply:
old habits die hard, don't they ? Well, moving the `*` to after `percent` is still 99% satisfactory to me, so it seems like a nice compromise !